### PR TITLE
Add QNX IDE, Makefile, and remove source code exec bit

### PR DESCRIPTION
**Pull request by JacobBarthelmeh**

### CODE REVIEW

**Comment by JacobBarthelmeh** [comment]
Builds off of https://github.com/wolfSSL/wolfssl/pull/6127

**Comment by embhorn** [comment]
Hi @JacobBarthelmeh 

Could you please rebase this and re-request a review?

File: IDE/QNX/README.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+WOLFSSL_DIR is used to point to the directory where wolfSSL was built. By default it looks for an adjacent 'wolfssl' directory. i.e.
+
+```
+mkdir test
+cd test
+git clone https://github.com/wolfssl/wolfssl
+git clone https://github.com/wolfssl/wolfmqtt
+```
+
+Then both projects can be imported as an momentics IDE project. To first build wolfSSL view the README located at wolfssl/IDE/QNX/README.md. Then to build wolfMQTT start momentics with the workspace located at wolfmqtt/IDE/QNX/. Import the wolfmqtt project (File->Import->General->Existing Projects into Workspace, in "Select root directory" browse to the directory wolfmqtt/IDE/QNX/ and select the wolfmqtt project then click "Finish".
```
Comment (embhorn) [inline IDE/QNX/README.md:10]: "an momentics" -> "a Momentics"

File: src/mqtt_packet.c
Comment (embhorn) [inline src/mqtt_packet.c:1]: Should this be removed from the review?

**Review by embhorn — CHANGES_REQUESTED** [review]
I did not build test. Do we have an automated build test for QNX / wolfSSL?

File: src/mqtt_packet.c
Comment (JacobBarthelmeh) [reply src/mqtt_packet.c:1]: it's a change for the execute bit on the file I noticed, it is outside of QNX use but just doing chmod 644 on the source file

**Review by embhorn — APPROVED** [review]

**Pull request merged**